Overrides: expose byType matcher UI

ryantxu opened the pull request:
This lets you define overrides byType

![image](https://user-images.githubusercontent.com/705951/90567043-59eb7500-e15e-11ea-8352-2b99b3d172ba.png)

This exposes a selector for *, string, number, time -- maybe we should only include the fields in the picker that have results?

ryantxu committed:
add type selectors

ryantxu committed:
add type selectors

ryantxu committed:
add type selectors

kylebrandt commented:
Works well from my testing with Azure. 

Main thing that confuses is me is that when selecting "Standard Options", you do not see there that an override is needed to change it, as they are blank even in the case where the datasource returns displayName or unit. Seems like we could use some sort of UI indicator there.

Although with overrides having matchers, I'm not sure I understand the point of "Standard Options", as I can not think of a scenario where it overrides would not also solve it.

ivanahuckova committed:
Loki: Add scopedVars support in legend formatting for repeated variables (#27046)

* Pass scoped vars to createMetricLabel

* Add tests

* Refacotr

* Update tests

* Update test

ivanahuckova committed:
Pass scoped  vars to create label (#27047)

hugohaggmark committed:
MixedDatasource: Shows retrieved data even if a data source fails (#27024)

* MixedDatasource: Shows data for working data sources and error for erroneous

* Tests: remove funky import

* Tests: fixes types

marcusolsson committed:
Add guide for sharing plugins (#27043)

* Add guide for sharing plugins

* Fix review comments

aknuds1 committed:
InfluxDB: Upgrade InfluxDB in devenv (#26983)

* InfluxDB: Upgrade InfluxDB in devenv
* InfluxDB: De-export symbols
* InfluxDB: Remove unused code
* devenv: Make InfluxDB version configurable

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Drone: Add Windows PR pipeline (#26589)

* Drone: Add Windows PR pipeline
* CI: Add Dockerfile for WiX image

aknuds1 committed:
Server: Add protocol to Cfg (#27071)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

kylebrandt committed:
Alerting: copy queryType prop to tsdb.Query (#27053)

fixes #27048

kaydelaney committed:
CloudWatch Logs: Move query response stats to appropriate FrameMeta property (#26732)

* CloudWatch Logs: Move query response stats to appropriate FrameMeta property

kylebrandt committed:
AzureAnalytics: Use fillmode Null with format as time series (#27054)

previously was using fillmode previous, which is not what we want here.
Also we do not have fill missing with this data source anyways
Fixes #27013

aknuds1 committed:
Drone: Fix order of steps (#27076)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

oddlittlebird committed:
Docs: Update troubleshoot-queries.md (#27078)

* Update troubleshoot-queries.md

* Update troubleshoot-queries.md

dprokop commented:
> This exposes a selector for *, string, number, time -- maybe we should only include the fields in the picker that have results?

Agree we should display only suggestion for the fields with results. Apart from that, I'm not sure about * selector. Isn't it the same as just specifying field option rather than override?

dprokop commented on packages/grafana-ui/src/components/MatchersUI/FieldTypeMatcherEditor.tsx:
```diff
@@ -0,0 +1,89 @@
+});
+
+const allTypes: Array<SelectableValue<string>> = [
+  { value: FieldType.number, label: 'Numeric fields' },
```
Nit: Numeric fields -> just Numeric ? Matcher name already suggests "fields"

dprokop reviewed (approved):
Think this looks good, some nits to consider:) @ryantxu Mind https://github.com/grafana/grafana/pull/27070 btw ;)

kylebrandt commented:
> I'm not sure about * selector. Isn't it the same as just specifying field option rather than override?

@dprokop Not presently. If a dataframe has Field Config information, then that currently takes precedence over the Field Options. So in this situation, the Field Options have no impact.

dprokop commented:
Ah yes, this makes sense then @kylebrandt

yesoreyeram committed:
plugin.json schema updates (#27029)

* added $schema field support in plugin.schema.json

More details can be found in https://github.com/grafana/plugin-validator/issues/1

* Others category in plugin.json, minimum 1 keyword 

* Allow others as a category if the plugin doesn't fall into appropriate category. Refer https://github.com/grafana/plugin-validator/issues/3 for more details
* Mandate minimum one keyword for better discovery in plugin repository.

* Dropped enum from plugin.json $schema

ryantxu committed:
remove type from the dropdown name

ryantxu commented:
Good point @dprokop -- updated with:
![image](https://user-images.githubusercontent.com/705951/90682432-c8d9d400-e219-11ea-924e-afde2b6f680f.png)

ryantxu committed:
Merge remote-tracking branch 'origin/master' into by-type-selectors

ryantxu commented:
Given that https://github.com/grafana/grafana/pull/27070 should also be in 7.1.4, I think we should remove the special case for "*" -- that makes more sense for something that is explicitly regex

ryantxu committed:
remove *

ryantxu committed:
don't count total
